src/workers/notifications/collectionUpdated.ts

```diff
@@ -0,0 +1,76 @@
+import { messageToJson } from '../worker';
+import {
+  NotificationPreferencePost,
+  Post,
+  PostRelation,
+  PostRelationType,
+  PostType,
+  Source,
+} from '../../entity';
+import { ChangeObject } from '../../types';
+import { NotificationHandlerReturn, NotificationWorker } from './worker';
+import { buildPostContext } from './utils';
+import { NotificationCollectionContext } from '../../notifications';
+import {
+  NotificationPreferenceStatus,
+  NotificationType,
+} from '../../notifications/common';
+
+interface Data {
+  post: ChangeObject<Post>;
+}
+
+export const collectionUpdated: NotificationWorker = {
+  subscription: 'api.post-collection-updated-notification',
+  handler: async (message, con) => {
+    const data: Data = messageToJson(message);
+
+    const baseCtx = await buildPostContext(con, data.post.id);
+    if (!baseCtx) {
+      return;
+    }
+    const { post } = baseCtx;
+
+    if (post.type !== PostType.Collection) {
+      return;
+    }
+
+    const notifs: NotificationHandlerReturn = [];
+
+    const distinctSources = await con
```

Comment:
Is this not already available directly on the postCollection technically?

Reply:
It is not since normalization takes place async in cdc so there could be the case where it is not normalized yet.

Reply:
That's good point, it is. We can reuse this fields

Reply:
@capJavert @omBratteng wouldn't it be better than to turn it into a function and reuse everywhere?

Reply:
I'm fine with either, I'll leave it up to @idoshamun

Reply:
@capJavert sounds great to me! Let's proceed with this idea

Reply:
@omBratteng if you can update that and we are good to go 🙏

Reply:
Working on it as we type

Reply:
Done, Also implemented it in the `normalizeCollectionPostSources` function. Just have to wait for tests to see if it works as intended 😄